### PR TITLE
Fix Op (scaled_dot_product_attention) | feat(torchlib)

### LINKED ISSUE #1799: [torchlib] Update `scaled_dot_product_attention` function signature
In https://github.com/pytorch/pytorch/pull/132689, the function is updated with an extra argument: enable_gqa

### PR DESCRIPTION
**Pull request by titaiwangms**
Fix #1799 

Add an extra argument: `enable_gqa` to unblock the export.
The real implementation: https://github.com/microsoft/onnxscript/issues/1802

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/nn.py [hunk 1]
```diff
@@ -2003,6 +2009,10 @@ def aten_scaled_dot_product_attention_bool_mask(
         is_causal and attn_mask is None
     ), "is_causal and attn_mask cannot be set at the same time"
 
+    assert (
+        not enable_gqa
+    ), "conversion of scaled_dot_product_attention not implemented if enable_gqa is True"
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/nn.py:2014]: Is it possible to implement? having partial implementation w/ assert is not recommended in torchlib

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/nn.py:2014]: I think GQA is now only accessible with contrib op though: https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.GroupQueryAttention

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/nn.py:2014]: Should be implementable as a function?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/nn.py:2014]: I guess likely the ONNX framework (but not ORT) users should have their own custom ops if they want to run a model with this op.

**Comment by justinchuby** [comment]
Ref: https://pytorch.org/docs/main/generated/torch.nn.functional.scaled_dot_product_attention.html

**Review by justinchuby — APPROVED** [review]
LGTM to unblock export

File: onnxscript/function_libs/torch_lib/ops/nn.py (hunk 1 above)
Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/nn.py:2014]: https://github.com/microsoft/onnxscript/issues/1802

**Pull request merged**